cmd/earthly/autocomplete.go

```diff
@@ -82,50 +83,80 @@ func (app *earthlyApp) autoCompleteImp(ctx context.Context) (err error) {
 
 	return err
 }
+
 func (app *earthlyApp) insertBashCompleteEntry() error {
+	u, err := user.Current()
+	if err != nil {
+		return errors.Wrapf(err, "could not get current user")
+	}
+	isRootUser := u.Uid == "0"
 	var path string
-	if runtime.GOOS == "darwin" {
-		path = "/usr/local/etc/bash_completion.d/earthly"
+	// Assume that non-root can't write to the system and that installation
+	// to root's home isn't desirable.  One possible exception might be if
+	// those directories are on an R/O filesystem, but user can install these
+	// manually in that case.
+	if isRootUser {
+		if runtime.GOOS == "darwin" {
+			path = "/usr/local/etc/bash_completion.d/earthly"
+		} else {
+			path = "/usr/share/bash-completion/completions/earthly"
+		}
 	} else {
-		path = "/usr/share/bash-completion/completions/earthly"
+		// https://github.com/scop/bash-completion/blob/master/README.md#faq
+		userPath, ok := os.LookupEnv("BASH_COMPLETION_USER_DIR")
+		if !ok {
+			// This will give a standardized fallback even if XDG isn't active
+			userPath = xdg.DataHome
+		}
+		path = filepath.Join(userPath, "bash-completion/completions/earthly")
+	}
+	ok, err := app.insertBashCompleteEntryAt(path)
+	if err == nil {
```

Comment:
can you move the `return err` from the end of this function here instead:
```
	if err != nil {
	    return err
	}
	if ok {
		...
	} else {
		...
	}
	return nil
```

It makes the code more robust to future changes where someone might unintentionally reassign to `err` before the final `return err` at the end of the function is hit.

Reply:
Sure thing; I'll make that change when I get a few minutes.